clang/include/clang/CIR/Dialect/Builder/CIRBaseBuilder.h

```diff
@@ -52,6 +52,15 @@ class CIRBaseBuilderTy : public mlir::OpBuilder {
     return cir::BoolAttr::get(getContext(), getBoolTy(), state);
   }
 
+  /// Create a for operation.
+  cir::ForOp createFor(
+      mlir::Location loc,
+      llvm::function_ref<void(mlir::OpBuilder &, mlir::Location)> condBuilder,
```

Comment:
Is this intentionally missing the init?  `ForStmt` has an Init, Condition, Step, and Body, but this handles only 3.

Reply:
That's an excellent question. The incubator (and this PR) emits the initialization in front of the `for` op, but it does seem like it is something we would want more tightly associated with the `for` in the emitted CIR.

Reply:
The initialization happens within the scope preceding the actual cir.for, etc. We have no intend of reproducing the AST 100% here, because of nested regions the alloca needs to be visible and dominate all the loop.